lib/schedule/helpers.ex

```diff
@@ -5,14 +5,25 @@ defmodule Schedule.Helpers do
       iex> Schedule.Helpers.merge_lists([[:b, :c], [:a, :b]])
       [:a, :b, :c]
 
-      iex> Schedule.Helpers.merge_lists([[:c, :a], [:a, :b], [:b, :c]])
-      [:c, :a, :b]
-
       iex> Schedule.Helpers.merge_lists([[:a, :b]])
       [:a, :b]
 
       iex> Schedule.Helpers.merge_lists([])
       []
+
+  If there's a cycle, gives preference to earlier inputs.
+
+      iex> Schedule.Helpers.merge_lists([[:a, :b], [:b, :c], [:c, :a]])
+      [:a, :b, :c]
+
+  Non-greedily avoids conflicts when possible.
+
+      iex> Schedule.Helpers.merge_lists([[:a, :b], [:a, :c], [:b, :c]])
+      [:a, :b, :c]
+
+      # Passing this test would require a completely new implementation
```

Comment:
I searched and found one. Route 11 has these route patterns:
```
11-3-0 ctypt dayo dotst       brway       sosta bdfch
11-2-0 ctypt dayo dotst       brway       tumed
11-3-1 ctypt dayo dotst cabgr brway waher tumed bdfch
11-2-1 ctypt dayo dotst cabgr brway waher tumed
```
but it results in `... brway sosta bdfch waher tumed` (note that `tumed` and `bdfch` are in the wrong order)
because the first time `tumed` and `bdfch` are merged you can't tell the order yet,
and then the algorithm is greedy so it can't undo that.

<img width="181" alt="Screen Shot 2020-10-20 at 10 17 57" src="https://user-images.githubusercontent.com/23065557/96599086-8c7c3000-12bd-11eb-99c5-100be6ed2ad5.png">

But I need to add a hint so that `sosta` shows up in the right place anway, so having a smarter algorithm wouldn't get us much.

I'm in the process of going through the 7 other routes that might also have this problem. There's few enough that I'll just add hints for them if needed.